internal/cloudapi/server.go

```diff
@@ -50,94 +53,87 @@ func NewServer(workers *worker.Server, rpmMetadata rpmmd.RPMMD, distros *distror
 // Create an http.Handler() for this server, that provides the composer API at
 // the given path.
 func (server *Server) Handler(path string, identityFilter []string) http.Handler {
-	r := chi.NewRouter()
+	e := echo.New()
 
 	if len(identityFilter) > 0 {
 		server.identityFilter = identityFilter
-		r.Use(server.VerifyIdentityHeader)
+		e.Use(server.VerifyIdentityHeader)
 	}
-	r.Use(server.IncRequests)
-	r.Route(path, func(r chi.Router) {
-		HandlerFromMux(server, r)
-	})
+	handler := apiHandlers{
+		server: server,
+	}
+	RegisterHandlers(e.Group(path, server.IncRequests), &handler)
 
-	return r
+	return e
 }
 
-func (server *Server) VerifyIdentityHeader(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func (server *Server) VerifyIdentityHeader(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
 		const identityHeaderKey contextKey = iota
 		type identityHeader struct {
 			Identity struct {
 				AccountNumber string `json:"account_number"`
 			} `json:"identity"`
 		}
-
-		idHeaderB64 := r.Header["X-Rh-Identity"]
+		idHeaderB64 := c.Request().Header.Get("X-Rh-Identity") //r.Header["X-Rh-Identity"]
 		if len(idHeaderB64) != 1 {
```

Comment:
Missed this one, since Header.Get() returns a string instead of a slice, len() != 1 won't work. `if idHeaderB64 == ""` => return error, otherwise continue.